scripts/jenkins/Jenkinsfile

```diff
@@ -849,7 +861,9 @@ def withDockerEnv(image, opts=[], cl) {
             timeout(activity: true, time: 5, unit: 'MINUTES') {
                 def cpu_count = cpuCount()
                 withEnv(["MAKEFLAGS='-j${cpu_count+2} -l${cpu_count*2}'",
-                         "CTEST_PARALLEL_LEVEL='${cpu_count+2}'"]) {
+                         "CTEST_PARALLEL_LEVEL='${cpu_count+2}'",
+                         "XDG_CONFIG_HOME=${WORKSPACE}/xdg/user",
+                         "XDG_CONFIG_DIRS=${WORKSPACE}/xdg/system"]) {
```

Comment:
It is confusing if the same environment is set for every build job but only used for some (or even one here). Isn't it possible to pass the env from the definition above (like PLUGINS and KDB_DEFAULT_RESOLVER)?

Reply:
those are fundamentally different settings.
PLUGINS, KDB_DEFAULT_RESOLVER = cmake flags
XDG_CONFIG_HOME, XDG_CONFIG_DIRS = environment variables

We could add a parameter to the buildAndTest function that sets those though

Reply:
Yes, that would be great.

Reply:
I tried to implement that now for 5hours. Saldy it is not that easy:
* WORKSPACE is not available at the point where we want to set the variables that reference it
* Lazy evaluation does not work because there is not even a WORKSPACE reference at that point
* templating the strings seemed like the way to go but apparently WORKSPACE is not really set for the environment (maybe only for shell variables?). I dug deep in the jenkins code but did not find a reason for why it is that way. Anyway no dice either.

As there is no problem with having the variables defined for all Docker Environments (they are simply not used in other cases) I will stop trying to implement something that Jenkins definitely does not want you to do.

While we could simply not use the default buildAndTest function I would argue that the gains are not worth having the additional function.

Reply:
I removed the commits moving the XDG configuration due to the reasons above. I kept the commits in https://github.com/ingwinlu/libelektra/tree/impl_xdg_debug.